client/state/reader/lists/actions.js

```diff
@@ -181,6 +187,98 @@ export function unfollowList( owner, slug ) {
 	};
 }
 
+/**
+ * Triggers a network request to update a list's details.
+ *
+ * @param  {Object}  list List details to save
+ * @return {Function} Action promise
+ */
+export function updateListDetails( list ) {
+	if ( ! list || ! list.owner || ! list.slug || ! list.title ) {
+		throw new Error( 'List owner, slug and title are required' );
+	}
+
+	const preparedOwner = decodeURIComponent( list.owner );
+	const preparedSlug = decodeURIComponent( list.slug );
+	const preparedList = Object.assign( {}, list, { owner: preparedOwner, slug: preparedSlug } );
+
+	return ( dispatch ) => {
+		dispatch( {
+			type: READER_LIST_UPDATE,
+			list
+		} );
+
+		return new Promise( ( resolve, reject ) => {
```

Comment:
Instead of explicitly making a new Promise, could we `then` the promise coming out of `readListsUpdate` and return the result of that? It would be a bit more in keeping with how folks generally chain Promises.